### PR TITLE
Modified Documentation for v2

### PR DESCRIPTION
**Pull request by ubdussamad**
This PR is now close to complete.

It's up for review. Minor changes might be required.

This PR is related to the Issue: #2223

### CODE REVIEW

File: docs/user-guide.rst [hunk 1]
```diff
@@ -46,11 +49,16 @@ HTTP verb:
...
 .. code-block:: python
 
-    >>> r = http.request(
-    ...     'POST',
-    ...     'http://httpbin.org/post',
-    ...     fields={'hello': 'world'}
-    ... )
+    import urllib3
+
+    r = urllib3.request(
```
Comment (sethmlarson) [inline docs/user-guide.rst]: The braces in this example don't line up horizontally, should be:

```python
resp = urllib3.request(
    ...
)
```

File: docs/user-guide.rst [hunk 2]
```diff
@@ -119,11 +136,17 @@ to ``False``. By default HTTP responses are closed after reading all bytes, this
...
+    import urllib3
+
+    http = urllib3.PoolManager() # Creating a seprate PoolManager instance.
+
+    r = http.request('GET', 'https://example.com', preload_content=False)
+    r.auto_close = False
+
+    for line in io.TextIOWrapper(r):
+        print(line)
+    #<!doctype html> ...
```
Comment (sethmlarson) [inline docs/user-guide.rst:155]: Let's remove the `# ...` here only because it's kinda clunky since we're in a loop.

File: docs/user-guide.rst [hunk 3]
```diff
@@ -20,23 +20,26 @@ First things first, import the urllib3 module:
...
 To make a request use :meth:`~poolmanager.PoolManager.request`:
 
 .. code-block:: python
 
-    >>> r = http.request('GET', 'http://httpbin.org/robots.txt')
-    >>> r.data
-    b'User-agent: *\nDisallow: /deny\n'
+    import urllib3
+    # Send the GET request.
+    r = urllib3.request('GET', 'https://httpbin.org/robots.txt')
```
Comment (sethmlarson) [inline docs/user-guide.rst]: May as well use proper names like `resp` instead of `r` if we're modifying the example

File: docs/user-guide.rst [hunk 4]
```diff
@@ -20,23 +20,26 @@ First things first, import the urllib3 module:
...
+    # Print the returned data.
+    print(r.data)
```
Comment (sethmlarson) [inline docs/user-guide.rst]: Let's put a blank line before every section so it's easier to parse visually

File: docs/user-guide.rst [hunk 5]
```diff
@@ -20,23 +20,26 @@ First things first, import the urllib3 module:
...
+    #b'User-agent: *\nDisallow: /deny\n'
```
Comment (sethmlarson) [inline docs/user-guide.rst:45]: Should be a space after every `#` on a comment line and some whitespace between every section. Also quotes should be Black formatted so double quotes. This example should look like this:

```python
    import urllib3

    # Creating a PoolManager instance for sending requests.
    http = urllib3.PoolManager()

    # Send the GET request and getting back a HTTPResponse object.
    resp = http.request("GET", "https://httpbin.org/robots.txt")

    # Print the returned data.
    print(resp.data)
    # b"User-agent: *\nDisallow: /deny\n"
```

**Review by sethmlarson — CHANGES_REQUESTED** [review]
Thank you! Left a handful of comments to start off with, apply these across the board and I'll review again :)

File: docs/user-guide.rst (hunk 3 above)
Comment (ubdussamad) [reply docs/user-guide.rst]: Yeah, did that in the latest commit

File: docs/user-guide.rst (hunk 2 above)
Comment (ubdussamad) [reply docs/user-guide.rst:155]: Done.

File: docs/user-guide.rst (hunk 5 above)
Comment (ubdussamad) [reply docs/user-guide.rst:45]: Will do.

**Comment by sethmlarson** [comment]
I pushed my own change, take a look at it and learn how the style should look for future PRs. If you are going to make more changes you'll have to do a `git pull` of the branch to grab my change too.

File: src/urllib3/util/url.py [hunk 6]
```diff
@@ -149,12 +149,18 @@ def url(self) -> str:
...
-            ... '/path', 'query', 'fragment').url
-            'http://username:password@host.com:80/path?query#fragment'
+            import urllib3
+
+            U = urllib3.util.parse_url("https://google.com/mail/")
+
+            print(U.url)
+            # "http://google.com/mail/"
+
+            print( urllib3.util.Url("http", "username:password",
```
Comment (ubdussamad) [inline src/urllib3/util/url.py:159]: I am totally not sure about the best way to format this.

File: docs/contributing.rst [hunk 7]
```diff
@@ -93,6 +93,50 @@ further parameterize pytest for local testing.
 For all valid arguments, check `the pytest documentation
 <https://docs.pytest.org/en/stable/usage.html#stopping-after-the-first-or-n-failures>`_.
 
+Documentation Guidelines
```
Comment (ubdussamad) [inline docs/contributing.rst:96]: Check this out, I am not sure if this is what you wanted.

**Review by ubdussamad — COMMENTED** [review]
Please check this out whenever you get time. Sorry, I know it's a lot to review at once. I tried the best I could but it'll likey have some errors.

**Comment by shazow** [comment]
Wish there was a way to preview a rendered version of this automagically, Netlify has spoiled me. Looks alright so far. :)

**Comment by sethmlarson** [comment]
@shazow There is though! https://urllib3--2231.org.readthedocs.build/en/2231

**Comment by shazow** [comment]
@SethMichaelLarson Whoa the future is here! Thank you.

Personally I'm not a huuuuge fan of how the new formatting looks. I think it's okay and I wouldn't block it, but I would propose maybe we keep the "repl" format for the first opening block perhaps? And do the copypasta format for the rest.

image

vs

image

It could just be my brain but the copypasta format for the intro block looks somehow staggered and harder to grok the intent of what it's trying to say.

The rest of the blocks, in user guide etc, actually look quite comprehensible and nice.

**Comment by ubdussamad** [comment]
> @SethMichaelLarson Whoa the future is here! Thank you.
> 
> ~Personally I'm not a huuuuge fan of how the new copypasta snippets feel in all cases. I think it's okay and I wouldn't block it, but~ (Actually I quite like the new formatting everywhere except the intro example.)
> 
> I would propose maybe we keep the "repl" format for the first opening block perhaps? And do the copypasta format for the rest.
> 
> image
> 
> vs
> 
> image
> 
> It could just be my brain but the copypasta format for the intro block looks somehow staggered and harder to grok the intent of what it's trying to say.
> 
> The rest of the blocks, in user guide etc, actually look quite comprehensible and nice.

@shazow, Indeed the console-style looks more vibrant. But I really don't have any opinion for/against its use in the docs. I like it either way.

Btw, do you want both the github readme & the front page of docs? Or just the github readme.

**Comment by ubdussamad** [comment]
@sethmlarson @shazow  Okay, Ill push those changes to both the files.

**Review by sethmlarson — APPROVED** [review]
This looks good to me now, thanks for all of this! :tada:

**Pull request merged**